### ISSUE TITLE
Assisted workflow

### ISSUE THREAD

**Issue by TobiasKletter** [tischi/spindle3d#20]
This is an older idea that re-surfaced while looking at _Xenopus_ extract spindles. These rarely have disc-shaped metaphase plate, so the axis determination fails and all measurements except the segmentations fail:

![image](https://user-images.githubusercontent.com/45814595/109664958-0080eb80-7b6e-11eb-8fcd-115b00da06c6.png)
![image](https://user-images.githubusercontent.com/45814595/109665211-3aea8880-7b6e-11eb-8242-98f22a795a63.png)
![image](https://user-images.githubusercontent.com/45814595/109665458-79804300-7b6e-11eb-8eee-0066f3b18e76.png)

So to circumvent this, users should be able to decouple the axis alignment from the shape of the chromatin, with the tradeoff of having to do some extra work:

- Before the analysis, users manually specify spindle poles using mulit-point ROIs in Fiji and save these
- In the analysis, these are then used to determine the main spindle axis and for alignment of the coordinate system
- The remaining computations could run as before

I think if this is implemented, the scope and applicability are widened considerably, without having to adapt the workflow much?

**Comment by tischi** [comment]
ping @manerotoni

**Comment by tischi** [comment]
@manerotoni @TobiasKletter 
The easiest and best for batch processing could be if the input data contained two point ROIs.
Then, in the code, we could

```
if ( inputImage.contains( two point ROIs ) )
{
     // use them as spindle poles and also as DNA axis
}
else
{
     // do the usual
}
```

**Comment by TobiasKletter** [comment]
@tischi  Agreed, I will upload some files with saved ROIs

**Comment by TobiasKletter** [comment]
https://www.dropbox.com/s/0e72o28redjt3xc/Bl_1_pointROIs.tif?dl=0
https://www.dropbox.com/s/o418s7nbubis6w9/S5-1_pointROIs.tif?dl=0 

Channel 1 = DNA
Channel 2 = spindle

**Comment by manerotoni** [comment]
Dear Tischi,
yes in fact in many organisms the metaphase plate is not cleary visible. I have just an example where the metaphase plate is more like an ellipsoid with 2 equal axes and one long axes. Also sometimes not all chromosome are aligned. There are phenotypes of misaligned chromosomes with still a properly formed spindle. Honestly sometimes if you lack the tubulin channel it is even impossible to figure out an axis. 

I still have not talked to Nicolas Chiarottini, but it is in the same line of thought of finding your axis of reference. 

Antonio

**Comment by tischi** [comment]
@manerotoni If we implement a spindle realignment based on two manually placed point ROIs into this plugin, then you would not need to "speak with Nico". I guess, my point is that I may now anyway do it and then you could just use it. Sounds good for you? Or would you like to have a standalone implementation in BDV?

**Comment by manerotoni** [comment]
Not sure. I think it is not only a spindle problem. Would you implement it in a way that one could export an imagePlus?
Also i think 2 points are not enough you need the plane so 3 points. For a confocal setup one could choose as a third point the point that is in the plane most orthogonal to Z. For a light-sheet isotropic data this may not  be the best choice. One should specify the 3rd point. 
the mails I get them on a different account. I need to redirect it.

**Comment by manerotoni** [comment]
Having said that even a default 2 point realignment would be great!

**Comment by tischi** [comment]
> Not sure. I think it is not only a spindle problem. Would you implement it in a way that one could export an imagePlus?

Yes, this would initially only be for spindles, because it will also compute a bunch of spindle specific properties **and output an aligned imagePlus** (note: the input **must** be an image with both DNA and tubulin staining).

I will ping you once it is working then you could try it out.

But, yes, for your general use case implementing something in BDV would be better.

**Comment by tischi** [comment]
@TobiasKletter 
The example image I worked with also contains bright DNA spots outside of the cell and thus the DNA segmentation fails ( think this was using the first of the example images ).

<img width="545" alt="image" src="https://user-images.githubusercontent.com/2157566/110322112-eaae7300-8012-11eb-8d29-bd38e5439e2d.png">

What should we do?

Can you find one which does not have this issue?

An alternative could be to somehow use the manually given spindle poles to create a rough cell mask (or use the spindle pole center as a rough dna mask center), but that could be a little bit of work and we would have to make some assumptions how the spindle length corresponds to the DNA width.

**Comment by tischi** [comment]
@TobiasKletter The second example that you posted did not have this issue, but the DNA segmentation did not work very well, because some regions were much dimmer than usual (in other words the intensity differences within the metaphase plate were larger than in our usual setting).

<img width="412" alt="image" src="https://user-images.githubusercontent.com/2157566/110331061-dff9db00-801e-11eb-838f-ea63794bbc9c.png">

Not sure what to do here.

**Comment by TobiasKletter** [comment]
@tischi 
Regarding the first sample:
Yes sorry that was a problematic sample

> An alternative could be to somehow use the manually given spindle poles to create a rough cell mask (or use the spindle pole center as a rough dna mask center), but that could be a little bit of work and we would have to make some assumptions how the spindle length corresponds to the DNA width.

In general I think this could be a useful addition, so the assisted workflow could also cope with these images. Maybe with the idea that spindle length (as determined by the user) * 1.5 (approximately...) would correspond to the cell diameter and thus to the ROI in which the DNA is expected. 

Here is a more standard spindle with point ROIs:
https://www.dropbox.com/s/e2rkeyci55y4rrm/CDK5RAP2_-1_pointROIs.tif?dl=0 

> The second example that you posted did not have this issue, but the DNA segmentation did not work very well, because some regions were much dimmer than usual (in other words the intensity differences within the metaphase plate were larger than in our usual setting).

It's a bit difficult to see. The DNA is these samples is generally a bit tricky because it sometimes consists of loosely-connected irregular patches, so I would not expect the DNA-based morphometry such as length and width to be absolutely bullet proof here. However, it would be nice if the spindle thresholding still works even when the chromatin mask isn't perfect, and thus the downstream spindle-based measurements are correct. 

I am also wondering whether for these samples it would be helpful to use a different voxel size for analysis?

**Comment by TobiasKletter** [comment]
@tischi 
Sorry this slipped my attention yesterday, but when I analyse an image with saved point ROIs, they seem to be ignored.

Raw image without annotated poles: 
https://www.dropbox.com/s/cdaniha1h9y5j0o/CDK5RAP2_-1.tif?dl=0

Same image but with intentionally misplaced poles:
https://www.dropbox.com/s/c6uokcygryw2ohw/CDK5RAP2_-1_pointROIs_wrong.tif?dl=0 

Both give the same output

**Comment by tischi** [comment]
Yes, I disabled it for now on purpose because I think we decided that this is not part of the publication. I suggest we can discuss adding this new feature after publication.

**Comment by TobiasKletter** [comment]
Ah ok, because I implemented it already in the user guide as a solution for "bad" analyses...

**Comment by tischi** [comment]
I would remove it, we have to think more carefully, e.g. what to do with the maxMetaphasePlateLength in such cases.

**Comment by TobiasKletter** [comment]
Ok then I will remove it for now from the wiki as well
